internal/v4/api.go

```diff
@@ -131,14 +131,15 @@ func ResolveURL(store *charmstore.Store, url *charm.Reference) error {
 	if url.Series != "" && url.Revision != -1 {
 		return nil
 	}
-	urls, err := store.ExpandURL(url)
-	if err != nil {
-		return errgo.Notef(err, "cannot expand URL")
+	entity, err := store.FindBestEntity(url, "_id", "promulgated-url")
+	if err != nil && errgo.Cause(err) != params.ErrNotFound {
```

Comment:
This leads to inconsistent error messages depending on how the ID was queried. I'm going to leave this.